### PR TITLE
Fix phone validation in registration page

### PR DESCRIPTION
**Pull request by akshay-testpress**
### Changes
- Used country code picker library to validate phone number with country code and fixed regex for a normal phone number.
### Reason for the changes
- Phone number validator was earlier passing only 10 digit numbers. Due this other countries were facing the issue in putting their phone number for registration.
### Stats
Number of Test Cases - 1

### CODE REVIEW

File: app/src/main/java/in/testpress/testpress/authenticator/RegisterActivity.java [hunk 1]
```diff
@@ -238,11 +244,16 @@ private boolean validate(){
            }
             if (verificationMethod.equals(MOBILE)) {
                 //Phone number verification
-                Pattern phoneNumberPattern = Pattern.compile("\\d{10}");
-                Matcher phoneNumberMatcher = phoneNumberPattern.matcher(phoneText.getText()
-                        .toString().trim());
-                if (!phoneNumberMatcher.matches()) {
-                    phoneText.setError("Please enter 10 digit valid mobile number");
+                boolean isPhoneValid;
```
Comment (waranlogesh) [inline app/src/main/java/in/testpress/testpress/authenticator/RegisterActivity.java]:
```suggestion
                boolean isPhoneNumberValid;
```

File: app/src/main/java/in/testpress/testpress/util/Validations.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
+package in.testpress.testpress.util;
+
+import com.hbb20.CountryCodePicker;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class Validations {
```
Comment (waranlogesh) [inline app/src/main/java/in/testpress/testpress/util/Validations.java]:
```suggestion
public class PhoneNumberValidator {
```

File: app/src/test/java/in/testpress/testpress/util/ValidationsTest.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+
+
+@RunWith(PowerMockRunner.class)
+public class ValidationsTest {
```
Comment (waranlogesh) [inline app/src/test/java/in/testpress/testpress/util/ValidationsTest.java]:
```suggestion
public class PhoneNumberValidatorTests {
```

File: app/src/test/java/in/testpress/testpress/util/ValidationsTest.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+    @Test
+    public void test_phone_number_validation() {
```
Comment (waranlogesh) [inline app/src/test/java/in/testpress/testpress/util/ValidationsTest.java]: MethodName_ExpectedBehavior_StateUnderTest

File: app/src/main/java/in/testpress/testpress/authenticator/RegisterActivity.java (hunk 1 above)
Comment (akshay-testpress) [reply app/src/main/java/in/testpress/testpress/authenticator/RegisterActivity.java]: Changed

File: app/src/test/java/in/testpress/testpress/util/ValidationsTest.java (hunk 3 above)
Comment (akshay-testpress) [reply app/src/test/java/in/testpress/testpress/util/ValidationsTest.java]: Changed

File: app/src/test/java/in/testpress/testpress/util/ValidationsTest.java (hunk 4 above)
Comment (akshay-testpress) [reply app/src/test/java/in/testpress/testpress/util/ValidationsTest.java]: Changed test name according to the convention asked.

**Pull request merged**